### ISSUE TITLE
Outdated XenTools installation for CoreOS

### ISSUE THREAD

**Issue by ruskofd** [xcp-ng/xcp#183]
The installation of XenTools for CoreOS is not working anymore, there are two problem in the installation script.

* The os-release name has changed from `CoreOS` to `Container Linux by CoreOS`.  Here is the output of the script :
```
Detected `Container Linux by CoreOS 2079.3.0 (Rhyolite)' (Container Linux by CoreOS version 2079).

Unable to install guest packages for distribution
Container Linux by CoreOS 2079.3.0 (Rhyolite) (Container Linux by CoreOS).

Unknown Linux distribution `Container Linux by CoreOS'.

You should manually install a version of
xe-guest-utilities which is suitable for your
distribution.
```

* The installation function try to copy a file that doesn't exist anymore in the XenTools ISO file.

```
install_coreos()
{
    echo "Installing the agent..."
    mkdir -p /usr/share/oem/xs/
    cp -f ${DATADIR}/xe-daemon \
          ${DATADIR}/xe-linux-distribution \
          ${DATADIR}/xe-update-guest-attrs /usr/share/oem/xs/ ### THIS IS THIS FILE (xe-update-guest-attrs)
    cp -f ${DATADIR}/xen-vcpu-hotplug.rules /etc/udev/rules.d/
    cp -f ${DATADIR}/xe-linux-distribution.service /etc/systemd/system/
    systemctl enable /etc/systemd/system/xe-linux-distribution.service
    echo "Installation complete."
    systemctl start xe-linux-distribution.service
}
```
The content of XCP-ng guest tool iso file :
```
 17K -r-xr-xr-x. 1 root root  17K Aug 24  2018 install.sh
 512 -r--r--r--. 1 root root  146 Mar 22  2018 versions.deb
 512 -r--r--r--. 1 root root  242 Mar 22  2018 versions.rpm
 512 -r--r--r--. 1 root root  143 Mar 22  2018 versions.tgz
1.5M -r-xr-xr-x. 1 root root 1.5M Aug 24  2018 xe-daemon
638K -r--r--r--. 1 root root 638K Mar 22  2018 xe-guest-utilities-7.10.0-1.i386.rpm
638K -r--r--r--. 1 root root 638K Mar 22  2018 xe-guest-utilities-7.10.0-1.x86_64.rpm
505K -r--r--r--. 1 root root 505K Mar 22  2018 xe-guest-utilities-xenstore-7.10.0-1.i386.rpm
505K -r--r--r--. 1 root root 505K Mar 22  2018 xe-guest-utilities-xenstore-7.10.0-1.x86_64.rpm
1.2M -r--r--r--. 1 root root 1.2M Mar 22  2018 xe-guest-utilities_7.10.0-1_all.tgz
1.2M -r--r--r--. 1 root root 1.2M Mar 22  2018 xe-guest-utilities_7.10.0-1_amd64.deb
1.2M -r--r--r--. 1 root root 1.2M Mar 22  2018 xe-guest-utilities_7.10.0-1_i386.deb
 14K -r-xr-xr-x. 1 root root  14K Aug 24  2018 xe-linux-distribution
 512 -r--r--r--. 1 root root  209 Aug 24  2018 xe-linux-distribution.service
 512 -r--r--r--. 1 root root  114 Aug 24  2018 xen-vcpu-hotplug.rules
```
Here is a simple working fix for this issue :
```
install_coreos()
{
    echo "Installing the agent..."
    mkdir -p /usr/share/oem/xs/
    cp -f ${DATADIR}/xe-daemon \
          ${DATADIR}/xe-linux-distribution /usr/share/oem/xs/
    cp -f ${DATADIR}/xen-vcpu-hotplug.rules /etc/udev/rules.d/
    cp -f ${DATADIR}/xe-linux-distribution.service /etc/systemd/system/
    systemctl enable /etc/systemd/system/xe-linux-distribution.service
    echo "Installation complete."
    systemctl start xe-linux-distribution.service
}
```

**Comment by stormi** [comment]
Thanks for the report. Can you contribute a pull request upstream to https://github.com/xenserver/xe-guest-utilities ?

I'll try to include the patch in XCP-ng for version 8.0.0.

**Comment by ruskofd** [comment]
I forget to mention that the issue is concerning only the "install.sh" bash script. So I can't bring my modifications since this script is not present in the repo.

**Comment by stormi** [comment]
Understood. There's no need to update `xe-linux-distribution`? (which is in the repo)

**Comment by ruskofd** [comment]
I don't think so. Because when I did my tests (5-6 times), I only delete the `xe-update-guest-attrs` string from the `cp` command line.

And concerning the problem with the distribution name (`Container Linux by CoreOS`), I only force the installation by typing the following command : `./install.sh -d CoreOS -m 2079` (the version wont impact but this is required). I think there only a single string to edit in the script also.

**Comment by stormi** [comment]
Could you try this version of install.sh and tell me if it works? https://github.com/xcp-ng/xe-guest-utilities/blob/master/mk/install.sh

**Comment by ruskofd** [comment]
I made some tests with the script you provided just above :

* **Detection of the OS** : good
```
Detected `Container Linux by CoreOS 2079.5.1 (Rhyolite)' (Container Linux by CoreOS version 2079).

The following changes will be made to this Virtual Machine:
  * update arp_notify sysctl.
  * packages to be installed/upgraded:
    - xe-linux-distribution

Continue? [y/n] 
```
```
Installing the agent...
Created symlink /etc/systemd/system/multi-user.target.wants/xe-linux-distribution.service → /etc/systemd/system/xe-linux-distribution.service.
Installation complete.
You should now reboot this Virtual Machine.
```
* **Installation of the differents files** : good
```
core@core ~/Linux $ sudo ls /usr/share/oem/xs/
xe-daemon  xe-linux-distribution
```

* **Status of the xe-linux-distribution** : good
```
● xe-linux-distribution.service - Linux Guest Agent
   Loaded: loaded (/etc/systemd/system/xe-linux-distribution.service; enabled; vendor preset: disabled)
   Active: active (running) since Wed 2019-06-12 12:09:10 UTC; 29s ago
  Process: 581 ExecStartPre=/usr/share/oem/xs/xe-linux-distribution /var/cache/xe-linux-distribution (code=exited, status=0/SUCCES>
 Main PID: 620 (xe-daemon)
    Tasks: 9 (limit: 30975)
   Memory: 8.5M
   CGroup: /system.slice/xe-linux-distribution.service
           ├─620 /usr/share/oem/xs/xe-daemon
           ├─624 logger -t xe-daemon
           └─628 logger -t xenstore

Jun 12 12:09:11 core systemd[1]: Starting Linux Guest Agent...
Jun 12 12:09:10 core systemd[1]: Started Linux Guest Agent.
```

And Xen Orchestra detecting that the PVHVM mode is activated.

**Comment by stormi** [comment]
I have pushed a new version of the xcp-ng-pv-tools RPM with that updated install.sh. It should reach the mirrors soon.

**Comment by stormi** [comment]
Reopened, let's backport that fix 7.6.

**Comment by stormi** [comment]
Updated pv tools RPM available in `xcp-ng-updates_testing` for 7.6.

Install with `yum update xcp-ng-pv-tools --enablerepo=xcp-ng-updates_testing`

Can someone confirm the fix?
